app/controllers/teams/joins_controller.rb

```diff
@@ -0,0 +1,17 @@
+class Teams::JoinsController < TeamsController
+  skip_before_action :find_team
+
+  def show
+    team = Team.find_by!(token: params[:team_token], url_join_allowed: true)
+
+    @team_membership = TeamMembership.new(team: team)
+  end
+
+  def create
+    team = Team.find_by!(token: params[:team_token], url_join_allowed: true)
+
+    TeamMembership.create!(team: team, user: current_user)
```

Comment:
Can you wrap this in an exception handler and redirect back to `:show` if it fails pls. (An example of why this might fail is if someone double clicks, so the first one creates the team and the second one then blows up, but that second response is the response the person gets. With this `rescue` and the guard above we avoid that scenario)